app/src/main/java/cm/aptoide/pt/app/view/AppViewFragment.java

```diff
@@ -1699,6 +1720,45 @@ private void showAppcInfo(boolean hasAdvertising, boolean hasBilling, double app
     }
   }
 
+  private void setCountdownTimer(String date) {
+    SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    DecimalFormat countdownNumberFormat = new DecimalFormat("00");
+    dateFormatter.setLenient(false);
+    final long now = System.currentTimeMillis();
+    long dateMillis = 0;
+    Date endDate;
+    try {
+      endDate = dateFormatter.parse(date);
+      dateMillis = endDate.getTime();
+    } catch (ParseException e) {
+      e.printStackTrace();
+    }
+
+    long timeToDisplay = dateMillis - now;
```

Comment:
I would extract this part related to the calculation of the timeToDisplay to another method to improve readability.